packages/graphql-codegen-cli/src/codegen.ts

```diff
@@ -125,7 +129,7 @@ export const validateCliOptions = (options: CLIOptions) => {
     logger.warn(`--export is deprecated, use --schema instead.`);
   }
 
-  if (!template && !project) {
+  if (!template && !project && !introspect) {
```

Comment:
avoid throwing an error since we don't need `template` or `project` while dumping a schema